correlation_rules/aws_sso_access_token_retrieved_by_unauthenticated_ip.yml

```diff
@@ -0,0 +1,30 @@
+AnalysisType: correlation_rule
+RuleID: "AWS.SSO.Access.Token.Retrieved.by.Unauthenticated.IP"
+DisplayName: "AWS SSO Access Token Retrieved by Unauthenticated IP"
+Enabled: true
+Severity: Medium
+Description: |-
+    When using AWS in an enterprise environment, best practices dictate to use a single sign-on service for identity and access management. AWS SSO is a popular solution, integrating with third-party providers such as Okta and allowing to centrally manage roles and permissions in multiple AWS accounts.
+
+    In this post, we demonstrate that AWS SSO is vulnerable by design to device code authentication phishing – just like any identity provider implementing OpenID Connect device code authentication. This technique was first demonstrated by Dr. Nestori Syynimaa for Azure AD. The feature provides a powerful phishing vector for attackers, rendering ineffective controls such as MFA (including Yubikeys) or IP allow-listing at the IdP level.
+Reference: https://blog.christophetd.fr/phishing-for-aws-credentials-via-aws-sso-device-code-authentication/
+Detection:
+    - Sequence: # Create a list of rules or signals to correlate (order matters)
+        - ID: SIGNAL-Sign-inwithAWSCLIprompt
+          RuleID: SIGNAL.-.Sign-in.with.AWS.CLI.prompt
+          Absence: true # makes sure the step did not happen
+        - ID: SIGNAL-RetrieveSSOaccesstoken
+          RuleID: SIGNAL.-.Retrieve.SSO.access.token
+          MinMatchCount: 1
+      Transitions: # Add conditions for each transition
+        - ID: Match SIGNAL-Sign-inwithAWSCLIprompt and SIGNAL-RetrieveSSOaccesstoken on IP
+          From: SIGNAL-Sign-inwithAWSCLIprompt
+          To: SIGNAL-RetrieveSSOaccesstoken
+          WithinTimeFrameMinutes: 120 # set max amount of time for this transition to occur
```

Comment:
This value is twice as long as the lookback window - did you forget to increase the lookback window from the default 60?